metricflow/model/parsing/dir_to_model.py

```diff
@@ -237,19 +236,20 @@ def parse_config_yaml(
 
 def parse(  # type: ignore[misc]
     _type: Type[Union[DataSource, Metric, Materialization]],
-    ctx: ParsingContext,
     yaml_dict: Dict[str, Any],
     filename: str,
-    contents: str,
+    contents_by_line: List[str],
```

Comment:
I don't have enough context on the YAML parser to understand why this is necessary. Would you mind updating the commit message to include an explanation of this why we're making this change?

Reply:
So initially, this wasn't even being used, but we do something similar in the old parser where we do a `contents.splitline()` to get the yaml lines by rows and get the file slice with the `context.start_line:context.end_line`. But it feels like just passing the splitted content already will avoid us from having to do a split every call to `parse` since it is recursive and it'll get very repetitive. I'll try to amend the commit message!